web/views.py

```
@@ -6928,6 +6931,171 @@ def all_study_groups(request):
+@login_required
+def work_submission_detail(request: HttpRequest, submission_id: int) -> HttpResponse:
+    """View to see the details of a specific work submission and provide feedback."""
+    submission = get_object_or_404(WorkSubmission, pk=submission_id)
+
+    # Determine user role
+    is_student = request.user == submission.student
+    is_reviewer = request.user.profile.is_teacher
+
+    # Check if the user has access
+    if not (is_student or is_reviewer):
+        raise PermissionDenied("You do not have permission to view this submission.")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)